<?xml version="1.0" encoding="UTF-8"?>
<testng-results skipped="1" failed="3" ignored="0" total="9" passed="5">
  <reporter-output>
    <line>
      <![CDATA[<a href="https://www.google.com" target="blank">Screenshot link</a>]]>
    </line>
    <line>
      <![CDATA[<br>]]>
    </line>
    <line>
      <![CDATA[<a href="https://www.google.com" target="blank"><img height=100 width=100 src="C:\Users\Admin\Documents\VTO\M3.jpg"></a>]]>
    </line>
    <line>
      <![CDATA[<a href="https://www.google.com" target="blank">Screenshot link</a>]]>
    </line>
    <line>
      <![CDATA[<br>]]>
    </line>
    <line>
      <![CDATA[<a href="https://www.google.com" target="blank"><img height=100 width=100 src="C:\Users\Admin\Documents\VTO\M3.jpg"></a>]]>
    </line>
    <line>
      <![CDATA[<a href="https://www.google.com" target="blank">Screenshot link</a>]]>
    </line>
    <line>
      <![CDATA[<br>]]>
    </line>
    <line>
      <![CDATA[<a href="https://www.google.com" target="blank"><img height=100 width=100 src="C:\Users\Admin\Documents\VTO\M3.jpg"></a>]]>
    </line>
  </reporter-output>
  <suite name="Functional" duration-ms="135" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
    <groups>
      <group name="bvt">
        <method signature="Testcase2.validateTitles()[pri:0, instance:learningTestNG.Testcase2@6c49835d]" name="validateTitles" class="learningTestNG.Testcase2"/>
        <method signature="Testcase3.userLogout()[pri:3, instance:learningTestNG.Testcase3@614ddd49]" name="userLogout" class="learningTestNG.Testcase3"/>
        <method signature="Testcase3.userReg()[pri:1, instance:learningTestNG.Testcase3@614ddd49]" name="userReg" class="learningTestNG.Testcase3"/>
      </group> <!-- bvt -->
      <group name="functional">
        <method signature="Testcase.userReg()[pri:1, instance:learningTestNG.Testcase@23e028a9]" name="userReg" class="learningTestNG.Testcase"/>
      </group> <!-- functional -->
      <group name="smoke">
        <method signature="Testcase.userLogin()[pri:2, instance:learningTestNG.Testcase@23e028a9]" name="userLogin" class="learningTestNG.Testcase"/>
        <method signature="Testcase3.test3()[pri:4, instance:learningTestNG.Testcase3@614ddd49]" name="test3" class="learningTestNG.Testcase3"/>
        <method signature="Testcase3.userLogout()[pri:3, instance:learningTestNG.Testcase3@614ddd49]" name="userLogout" class="learningTestNG.Testcase3"/>
      </group> <!-- smoke -->
    </groups>
    <test name="Annotation Test" duration-ms="50" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
      <class name="learningTestNG.Testcase">
        <test-method status="PASS" signature="createDB()[pri:0, instance:learningTestNG.Testcase@23e028a9]" name="createDB" is-config="true" duration-ms="0" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- createDB -->
        <test-method status="PASS" signature="launchBrowser()[pri:0, instance:learningTestNG.Testcase@23e028a9]" name="launchBrowser" is-config="true" duration-ms="1" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- launchBrowser -->
        <test-method status="PASS" signature="userReg()[pri:1, instance:learningTestNG.Testcase@23e028a9]" name="userReg" duration-ms="8" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- userReg -->
        <test-method status="PASS" signature="closeBrowser()[pri:0, instance:learningTestNG.Testcase@23e028a9]" name="closeBrowser" is-config="true" duration-ms="0" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeBrowser -->
        <test-method status="PASS" signature="launchBrowser()[pri:0, instance:learningTestNG.Testcase@23e028a9]" name="launchBrowser" is-config="true" duration-ms="0" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- launchBrowser -->
        <test-method status="PASS" signature="userLogin()[pri:2, instance:learningTestNG.Testcase@23e028a9]" name="userLogin" duration-ms="2" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- userLogin -->
        <test-method status="PASS" signature="closeBrowser()[pri:0, instance:learningTestNG.Testcase@23e028a9]" name="closeBrowser" is-config="true" duration-ms="1" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeBrowser -->
        <test-method status="PASS" signature="launchBrowser()[pri:0, instance:learningTestNG.Testcase@23e028a9]" name="launchBrowser" is-config="true" duration-ms="1" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- launchBrowser -->
        <test-method status="PASS" signature="userLogout()[pri:3, instance:learningTestNG.Testcase@23e028a9]" name="userLogout" duration-ms="1" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- userLogout -->
        <test-method status="PASS" signature="closeBrowser()[pri:0, instance:learningTestNG.Testcase@23e028a9]" name="closeBrowser" is-config="true" duration-ms="0" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeBrowser -->
        <test-method status="PASS" signature="closeDB()[pri:0, instance:learningTestNG.Testcase@23e028a9]" name="closeDB" is-config="true" duration-ms="0" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- closeDB -->
      </class> <!-- learningTestNG.Testcase -->
    </test> <!-- Annotation Test -->
    <test name="Assertion Test" duration-ms="13" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
      <class name="learningTestNG.Testcase2">
        <test-method status="FAIL" signature="validateTitles()[pri:0, instance:learningTestNG.Testcase2@6c49835d]" name="validateTitles" duration-ms="8" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[The following asserts failed:
expected [Yahoo.com] but found [Gmail.com],
Image validation Failed expected [false] but found [true]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: The following asserts failed:
expected [Yahoo.com] but found [Gmail.com],
Image validation Failed expected [false] but found [true]
at org.testng.asserts.SoftAssert.assertAll(SoftAssert.java:43)
at learningTestNG.Testcase2.validateTitles(Testcase2.java:31)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
at org.testng.TestRunner.privateRun(TestRunner.java:648)
at org.testng.TestRunner.run(TestRunner.java:505)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
at org.testng.SuiteRunner.run(SuiteRunner.java:364)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
at org.testng.TestNG.runSuites(TestNG.java:1049)
at org.testng.TestNG.run(TestNG.java:1017)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:114)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
            <line>
              <![CDATA[<a href="https://www.google.com" target="blank">Screenshot link</a>]]>
            </line>
            <line>
              <![CDATA[<br>]]>
            </line>
            <line>
              <![CDATA[<a href="https://www.google.com" target="blank"><img height=100 width=100 src="C:\Users\Admin\Documents\VTO\M3.jpg"></a>]]>
            </line>
          </reporter-output>
        </test-method> <!-- validateTitles -->
      </class> <!-- learningTestNG.Testcase2 -->
    </test> <!-- Assertion Test -->
    <test name="SoftDependany Test" duration-ms="16" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
      <class name="learningTestNG.Testcase3">
        <test-method status="FAIL" signature="userReg()[pri:1, instance:learningTestNG.Testcase3@614ddd49]" name="userReg" duration-ms="3" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Registration failed]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Registration failed
at org.testng.Assert.fail(Assert.java:96)
at learningTestNG.Testcase3.userReg(Testcase3.java:12)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
at org.testng.TestRunner.privateRun(TestRunner.java:648)
at org.testng.TestRunner.run(TestRunner.java:505)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
at org.testng.SuiteRunner.run(SuiteRunner.java:364)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
at org.testng.TestNG.runSuites(TestNG.java:1049)
at org.testng.TestNG.run(TestNG.java:1017)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:114)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
            <line>
              <![CDATA[<a href="https://www.google.com" target="blank">Screenshot link</a>]]>
            </line>
            <line>
              <![CDATA[<br>]]>
            </line>
            <line>
              <![CDATA[<a href="https://www.google.com" target="blank"><img height=100 width=100 src="C:\Users\Admin\Documents\VTO\M3.jpg"></a>]]>
            </line>
          </reporter-output>
        </test-method> <!-- userReg -->
        <test-method status="SKIP" signature="userLogin()[pri:2, instance:learningTestNG.Testcase3@614ddd49]" name="userLogin" duration-ms="0" started-at="2020-09-15T17:36:12Z" depends-on-methods="learningTestNG.Testcase3.userReg" finished-at="2020-09-15T17:36:12Z">
          <exception class="java.lang.Throwable">
            <message>
              <![CDATA[Method Testcase3.userLogin()[pri:2, instance:learningTestNG.Testcase3@614ddd49] depends on not successfully finished methods]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.Throwable: Method Testcase3.userLogin()[pri:2, instance:learningTestNG.Testcase3@614ddd49] depends on not successfully finished methods
at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:888)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
at org.testng.TestRunner.privateRun(TestRunner.java:648)
at org.testng.TestRunner.run(TestRunner.java:505)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
at org.testng.SuiteRunner.run(SuiteRunner.java:364)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
at org.testng.TestNG.runSuites(TestNG.java:1049)
at org.testng.TestNG.run(TestNG.java:1017)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:114)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.Throwable -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- userLogin -->
        <test-method status="PASS" signature="userLogout()[pri:3, instance:learningTestNG.Testcase3@614ddd49]" name="userLogout" duration-ms="2" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- userLogout -->
        <test-method status="PASS" signature="test3()[pri:4, instance:learningTestNG.Testcase3@614ddd49]" name="test3" duration-ms="2" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- test3 -->
      </class> <!-- learningTestNG.Testcase3 -->
    </test> <!-- SoftDependany Test -->
    <test name="Failure Test" duration-ms="6" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
      <class name="learningTestNG.TestFailure">
        <test-method status="PASS" signature="setUp()[pri:0, instance:learningTestNG.TestFailure@51cdd8a]" name="setUp" is-config="true" duration-ms="17" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- setUp -->
        <test-method status="FAIL" signature="doLogin()[pri:0, instance:learningTestNG.TestFailure@51cdd8a]" name="doLogin" duration-ms="3" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
          <exception class="java.lang.AssertionError">
            <message>
              <![CDATA[Failing Login Test]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.AssertionError: Failing Login Test
at org.testng.Assert.fail(Assert.java:96)
at learningTestNG.TestFailure.doLogin(TestFailure.java:12)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
at java.lang.reflect.Method.invoke(Unknown Source)
at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
at org.testng.internal.Invoker.invokeMethod(Invoker.java:583)
at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:719)
at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:989)
at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
at org.testng.TestRunner.privateRun(TestRunner.java:648)
at org.testng.TestRunner.run(TestRunner.java:505)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
at org.testng.SuiteRunner.run(SuiteRunner.java:364)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
at org.testng.TestNG.runSuites(TestNG.java:1049)
at org.testng.TestNG.run(TestNG.java:1017)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:114)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:251)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:77)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.AssertionError -->
          <reporter-output>
            <line>
              <![CDATA[<a href="https://www.google.com" target="blank">Screenshot link</a>]]>
            </line>
            <line>
              <![CDATA[<br>]]>
            </line>
            <line>
              <![CDATA[<a href="https://www.google.com" target="blank"><img height=100 width=100 src="C:\Users\Admin\Documents\VTO\M3.jpg"></a>]]>
            </line>
          </reporter-output>
        </test-method> <!-- doLogin -->
        <test-method status="PASS" signature="tearDown()[pri:0, instance:learningTestNG.TestFailure@51cdd8a]" name="tearDown" is-config="true" duration-ms="0" started-at="2020-09-15T17:36:12Z" finished-at="2020-09-15T17:36:12Z">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- tearDown -->
      </class> <!-- learningTestNG.TestFailure -->
    </test> <!-- Failure Test -->
  </suite> <!-- Functional -->
</testng-results>
